maat-court-data-api/src/main/java/gov/uk/courtdata/prosecutionconcluded/service/ProsecutionConcludedService.java

```diff
@@ -1,16 +1,87 @@
 package gov.uk.courtdata.prosecutionconcluded.service;
 
+import gov.uk.courtdata.entity.WQHearingEntity;
+import gov.uk.courtdata.enums.JurisdictionType;
+import gov.uk.courtdata.exception.MAATCourtDataException;
+import gov.uk.courtdata.hearing.crowncourt.validator.CaseTypeValidator;
+import gov.uk.courtdata.hearing.crowncourt.validator.OUCodeValidator;
+import gov.uk.courtdata.model.crowncourt.OffenceSummary;
 import gov.uk.courtdata.model.crowncourt.ProsecutionConcluded;
+import gov.uk.courtdata.prosecutionconcluded.CalculateCrownCourtOutcome;
+import gov.uk.courtdata.prosecutionconcluded.dto.ConcludedDTO;
+import gov.uk.courtdata.repository.WQHearingRepository;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 @Service
 @RequiredArgsConstructor
 @Slf4j
 public class ProsecutionConcludedService {
 
+    private final CalculateCrownCourtOutcome calculateCrownCourtOutcome;
+
+    private final WQHearingRepository wqHearingRepository;
+
+    private final OUCodeValidator ouCodeValidator;
+    private final CaseTypeValidator caseTypeValidator;
+
+    private final ProsecutionConcludedDAO prosecutionConcludedDAO;
+
     public void execute(final ProsecutionConcluded prosecutionConcluded) {
 
+        WQHearingEntity wqHearingEntity = wqHearingRepository.getById(prosecutionConcluded.getProsecutionCaseId().toString());
```

Comment:
Are we persisting the case ID during a hearing call?  The look-up should be hearing ID-based.

Reply:
This will be based on the maat-id and hearing-id.